db/compaction/compaction_job.cc

```diff
@@ -614,14 +618,23 @@ Status CompactionJob::Run() {
   }
 
   IOStatus io_s;
+  for (const auto& state : compact_->sub_compact_states) {
+    if (!state.io_status.ok()) {
+      io_s = state.io_status;
+      break;
+    }
+  }
+  if (!io_s.ok()) {
+    io_status_ = io_s;
+    status = io_s;
```

Comment:
@anand1976 I'm not sure here, if io_s is not ok, it is possible that status is already set to not ok at L615. Should we also do status = io_s here?

Reply:
Looks like in other places ```status``` takes precedence over ```io_s```, so we should do the same here for consistency